### PR TITLE
Add workflow to enforce versioning labels

### LINKED ISSUE #1047: Add workflow to enforce versioning labels
- Any one of the following labels should be present `Patch`, `Minor`, `Major`.
- Version update should happen based on the label

@shamilsdq _a

### PR DESCRIPTION
**Pull request by shamilsdq**
Closes #1047 

- Added workflow to ensure exactly one of 'major', 'minor' or 'patch' label is set
- Set version upgrade type according to the presence of 'major', 'minor' or 'patch' label
- Set git tagging option according to the presence or absence of 'release' label

### CODE REVIEW

**Comment by shamilsdq** [comment]
@amaldinesh7

**Comment by shamilsdq** [comment]
Tagging option has been set according to the presence of release label

@amaldinesh7

**Review by amaldinesh7 — CHANGES_REQUESTED** [review]
@shamilsdq _a Please check the comment

File: .github/workflows/version-bumper.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+on:
+  pull_request:
+    branches:
+      - main
+    types: [closed]
+
+jobs:
+  check:
+    runs-on: ubuntu-latest
+    steps:
```
[the remark this answers is not in the document]
Comment (shamilsdq) [reply .github/workflows/version-bumper.yml]: Done

**Comment by shamilsdq** [comment]
Replaced old version bumping steps with new version bumping steps in publish.yml

**Review by amaldinesh7 — APPROVED** [review]

**Pull request merged**